src/Integration.Vsix.UnitTests/Analysis/IssueHandlerTests.cs

```diff
@@ -31,134 +31,181 @@
 
-            publisher.PublishedSnapshot.AnalyzedFilePath.Should().Be(expectedFilePath);
+            notificationHandler.UpdatedSnapshot.AnalyzedFilePath.Should().Be(expectedFilePath);
         }
 
         [TestMethod]
-        public void WhenNewIssuesAreFound_AndFilterRemovesAllIssues_ListenersAreUpdated()
+        public void HandleNewIssues_IssuesAreFiltered()
         {
-            // Arrange
-            var filteredIssue = CreateIssue("single issue", startLine: 1, endLine: 1);
+            var inputIssues = new[]
+            {
+                CreateIssue("S111", startLine: 1, endLine: 1),
+                CreateIssue("S222", startLine: 2, endLine: 2)
+            };
 
-            var issuesToReturnFromFilter = Enumerable.Empty<IAnalysisIssueVisualization>();
-            var issuesFilter = CreateIssuesFilter(out var capturedFilterInput, issuesToReturnFromFilter);
+            var issuesToReturnFromFilter = new[]
+            {
+                CreateIssue("xxx", startLine: 3, endLine: 3)
+            };
 
-            var publisher = new SnapshotPublisher();
-            var testSubject = CreateTestSubject(issuesFilter.Object, publisher.PublishSnapshot);
+            var issuesFilter = CreateIssuesFilter(issuesToReturnFromFilter);
+            var notificationHandler = new SnapshotChangeHandler();
+
+            var testSubject = CreateTestSubject(notificationHandler.OnSnapshotChanged, issuesFilter.Object);
 
             // Act
-            testSubject.HandleNewIssues(new[] { filteredIssue });
+            testSubject.HandleNewIssues(inputIssues);
 
             // Assert
-            // Check the expected issues were passed to the filter
-            capturedFilterInput.Count.Should().Be(1);
-            capturedFilterInput[0].Should().Be(filteredIssue);
+            issuesFilter.Verify(x => x.Filter(inputIssues), Times.Once);
+            notificationHandler.InvocationCount.Should().Be(1);
+
+            // Check the updated issues
+            notificationHandler.UpdatedSnapshot.Issues.Count().Should().Be(1);
+            notificationHandler.UpdatedSnapshot.Issues.Should().BeEquivalentTo(issuesToReturnFromFilter);
+        }
+
+        [TestMethod]
+        public void HandleNewIssues_SpansAreTranslated()
+        {
+            var inputIssues = new[]
+            {
+                CreateIssue("xxx", startLine: 1, endLine: 1)
+            };
 
-            publisher.InvocationCount.Should().Be(1);
-            publisher.PublishedSnapshot.Issues.Count().Should().Be(0);
+            var issuesToReturnFromTranslator = new[]
+            {
+                CreateIssue("yyy", startLine: 2, endLine: 2)
+            };
+
+            var issuesFilter = CreatePassthroughIssuesFilter();
+            var notificationHandler = new SnapshotChangeHandler();
+
+            var textDocument = CreateValidTextDocument("any.txt");
+
+            var translator = new Mock<TranslateSpans>();
```

Comment:
Btw, I just remembered that we have https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Infrastructure.VS/SpanTranslator.cs

Reply:
I'll leave it as it is for now.